### PR TITLE
DMD nonlinear elasticity

### PR DESCRIPTION
**Pull request by siuwuncheung**

### CODE REVIEW

**Comment by chldkdtn** [comment]
@siuwuncheung is this ready for review?

**Comment by siuwuncheung** [comment]
> @siuwuncheung is this ready for review?

The PR is RFR now.

File: examples/dmd/nonlinear_elasticity.py [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+
+   How to run:
+      mpirun -np 8 python <arguments>
+
+      Example 1 of arguments:
+      nonlinear_elasticity.py -s 2 -rs 1 -dt 0.01 -tf 5 -visit
+   
+      Output 1:
+      Relative error of DMD velocity (v) at t_final: 5.000000 is 1.145E-03
+      Relative error of DMD position (x) at t_final: 5.000000 is 7.371E-05
```
Comment (chldkdtn) [inline examples/dmd/nonlinear_elasticity.py]: I am getting following in quartz:

Relative error of DMD velocity (v) at t_final: 5.000000 is 2.501E-03
Relative error of DMD position (x) at t_final: 5.000000 is 1.325E-04

Have you run this in quartz as well?

File: examples/dmd/nonlinear_elasticity.py [hunk 2]
```diff
@@ -0,0 +1,629 @@
...
+
+      Example 2 of arguments:
+      nonlinear_elasticity.py -s 2 -rs 1 -dt 0.01 -tf 5 -nwinsamp 10 -visit
+   
+      Output 2:
+      Relative error of DMD velocity (v) at t_final: 5.000000 is 1.106E-04
+      Relative error of DMD position (x) at t_final: 5.000000 is 4.304E-06
```
Comment (chldkdtn) [inline examples/dmd/nonlinear_elasticity.py]: Again, I am getting the following in quartz, using 8 processes and pdebug:

Relative error of DMD velocity (v) at t_final: 5.000000 is 6.509E-04
Relative error of DMD position (x) at t_final: 5.000000 is 9.197E-06

**Review by chldkdtn — APPROVED** [review]

**Comment by chldkdtn** [comment]
@bumpwy @dreamer2368 @pravija12345 if you can review this, that would be wonderful.

**Review by dreamer2368 — APPROVED** [review]

**Pull request merged**